Foundation/Codable.swift

```diff
@@ -0,0 +1,105 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+//===----------------------------------------------------------------------===//
+// Errors
+//===----------------------------------------------------------------------===//
+
+// Adding the following extensions to EncodingError and DecodingError allows them to bridge to NSErrors implicitly.
+
+fileprivate let NSCodingPathErrorKey = "NSCodingPath"
```

Comment:
Since we don't have error bridging in Linux, I think this whole section (short of the helper methods on `DecodingError`) can be removed.